content/en/docs/reference/using-api/cel.md

```diff
@@ -0,0 +1,198 @@
+| 'string' with format=duration                      | duration (google.protobuf.Duration)                                                                                          |
+
+xref: [CEL types](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#values),
+[OpenAPI types](https://swagger.io/specification/#data-types),
+[Kubernetes Structural Schemas](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema).
+
+Equality on arrays with `x-kubernetes-list-type` of `set` or `map` ignores element
+order, i.e. `[1, 2] == [2, 1]`.
+
+Concatenation on arrays with` x-kubernetes-list-type` use the semantics of the
+list type:
+
+- 'set': `X + Y` performs a union where the array positions of all elements in
+  `X` are preserved and non-intersecting elements in `Y` are appended, retaining
+  their partial order.
+- 'map': `X + Y` performs a merge where the array positions of all keys in `X`
+  are preserved but the values are overwritten by values in `Y` when the key
+  sets of `X` and `Y` intersect. Elements in `Y` with non-intersecting keys are
+  appended, retaining their partial order.
+
+### Escaping
+
+Only Kubernetes resource property names of the form
+`[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible from CEL. Accessible property
+names are escaped according to the following rules when accessed in the
+expression:
+
+| escape sequence   | property name equivalent                                                                     |
+|-------------------|----------------------------------------------------------------------------------------------|
+| `__underscores__` | `__`                                                                                         |
+| `__dot__`         | `.`                                                                                          |
+| `__dash__`        | `-`                                                                                          |
+| `__slash__`       | `/`                                                                                          |
+| `__{keyword}__`   | [CEL RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax) |
+
+Note that CEL RESERVED keywords need to match the exact property name to be escaped
+(e.g. int in the word sprint would not be escaped).
+
+Examples on escaping:
+
+| property name | rule with escaped property name   |
+|---------------|-----------------------------------|
+| namespace     | `self.__namespace__ > 0`          |
+| x-prop        | `self.x__dash__prop > 0`          |
+| redact__d     | `self.redact__underscores__d > 0` |
+| string        | `self.startsWith('kube')`         |
+
+### Resource Constraints
+
+To ensure that CEL expressions are safe to be run directly in the API server,
```

Comment:
```suggestion
CEL is non-Turing complete and offers a variety of production safety controls to limit execution time. CEL's _resource constraint_ features provide feedback to developers about expression complexity and help protect the API server from excessive resource consumption during evaluation.
```

Reply:
This suggestion isn't acceptable as-is (committing it would yield bad wording) but I like the intent.